SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1836,45 +1916,63 @@ void USpatialNetDriver::TickDispatch(float DeltaTime)
 			return;
 		}
 
-		if (LoadBalanceEnforcer.IsValid())
-		{
-			SCOPE_CYCLE_COUNTER(STAT_SpatialUpdateAuthority);
-			LoadBalanceEnforcer->Advance();
-			// Immediately flush. The messages to spatial created by the load balance enforcer in response
-			// to other workers should be looped back as quick as possible.
-			Connection->Flush();
-		}
+		const bool bIsDefaultServerOrClientWorker = [this] {
+			if (IsServer())
+			{
+				USpatialGameInstance* GameInstance = GetGameInstance();
+				return GameInstance->GetSpatialWorkerType() == SpatialConstants::DefaultServerWorkerType;
+			}
+			// Assume client, since the GameInstance might not be around.
+			return true;
+		}();
 
-		if (RPCService.IsValid())
+		if (bIsDefaultServerOrClientWorker)
 		{
-			RPCService->AdvanceView();
-		}
+			if (LoadBalanceEnforcer.IsValid())
+			{
+				SCOPE_CYCLE_COUNTER(STAT_SpatialUpdateAuthority);
+				LoadBalanceEnforcer->Advance();
+				// Immediately flush. The messages to spatial created by the load balance enforcer in response
+				// to other workers should be looped back as quick as possible.
+				Connection->Flush();
+			}
 
-		if (DebugCtx != nullptr)
-		{
-			DebugCtx->AdvanceView();
-		}
+			if (DebugCtx != nullptr)
+			{
+				DebugCtx->AdvanceView();
+			}
 
-		if (ActorSystem.IsValid())
-		{
-			ActorSystem->Advance();
-		}
+			if (RPCService.IsValid())
+			{
+				RPCService->AdvanceView();
+			}
 
-		{
-			SCOPE_CYCLE_COUNTER(STAT_SpatialProcessOps);
-			Dispatcher->ProcessOps(GetOpsFromEntityDeltas(Connection->GetEntityDeltas()));
-			Dispatcher->ProcessOps(Connection->GetWorkerMessages());
-			CrossServerRPCHandler->ProcessMessages(Connection->GetWorkerMessages(), DeltaTime);
-		}
+			if (ActorSystem.IsValid())
```

Comment:
Some of these systems aren't initialized for the routing worker - maybe this path can be branchless with .IsValid checks

Reply:
Don't know if separating provides more context though, so just an observation

Reply:
Or maybe the logic could be seperated into something like TickDispatchRoutingWorker and TickDispatchServerClientWorker?

Reply:
Or a separate NetDriver? I don't know how much is actually shared though :D

Reply:
They are separated though. May be hard to read because the diff is bad, but this gated by if (bIsDefaultServerOrClientWorker)
I do agree though that this can be branchless because most of these will be initialized all the time, but that's another debate.